producer/nf_management.go

```diff
@@ -329,53 +346,59 @@ func sendNFDownNotification(nfProfile models.NfProfile, nfInstanceID string) {
 	}
 }
 
-func UpdateNFInstanceProcedure(nfInstanceID string, patchJSON []byte) (response map[string]interface{}) {
+func UpdateNFInstanceProcedure(nfInstanceID string, patchJSON []byte) (response map[string]interface{}, err error) {
+	// Validation for NF Instance ID
+	if nfInstanceID == "" {
+		logger.ManagementLog.Errorln("NF Instance ID is required")
+		return nil, fmt.Errorf("NF Instance ID is required")
+	}
 	collName := "NfProfile"
 	filter := bson.M{"nfInstanceId": nfInstanceID}
 
-	err := dbadapter.DBClient.RestfulAPIJSONPatch(collName, filter, patchJSON)
-	if err == nil {
-		nf, _ := dbadapter.DBClient.RestfulAPIGetOne(collName, filter)
-
-		nfProfilesRaw := []map[string]interface{}{
-			nf,
-		}
-
-		nfProfiles, decodeErr := util.Decode(nfProfilesRaw, time.RFC3339)
-		if decodeErr != nil {
-			logger.ManagementLog.Info(decodeErr.Error())
-		}
-
-		// Update expiry time for document.
-		// Currently we are using 3 times the hearbeat timer as the expiry time interval.
-		// We should update it to be configurable : TBD
-		if factory.NrfConfig.Configuration.NfProfileExpiryEnable {
-			timein := time.Now().Local().Add(time.Second * time.Duration(factory.NrfConfig.Configuration.NfKeepAliveTime*3))
-			nf["expireAt"] = timein
-		}
-		//dbadapter.DBClient.RestfulAPIJSONPatch(collName, filter, jsonStr)
-		if ok, _ := dbadapter.DBClient.RestfulAPIPutOne(collName, filter, nf); ok {
-			logger.ManagementLog.Infof("nf profile [%s] update success", nfProfiles[0].NfType)
-		} else {
-			logger.ManagementLog.Infof("nf profile [%s] update failed", nfProfiles[0].NfType)
-		}
+	// Patch the existing NF Instance
+	patchError := dbadapter.DBClient.RestfulAPIJSONPatch(collName, filter, patchJSON)
+	if patchError != nil {
+		logger.ManagementLog.Errorln("Patch error in UpdateNFInstanceProcedure: ", patchError)
+		return nil, fmt.Errorf("patch error: %v", patchError)
+	}
+	// Get the updated NF Instance
+	nf, getErr := dbadapter.DBClient.RestfulAPIGetOne(collName, filter)
+	if getErr != nil || nf == nil {
+		logger.ManagementLog.Errorln("Failed to get NF instance: ", getErr)
```

Comment:
When using xxxxxxln(), there is no need to leave an empty space in the string (e.g., "sdsdcdfdfs ") because there is a blank space already added between the arguments
```suggestion
		logger.ManagementLog.Errorln("failed to get NF instance:", getErr)
```